ygen/genir.go

```diff
@@ -53,6 +53,13 @@ type IROptions struct {
 	// to true.
 	// NOTE: This flag will be removed by v1 release.
 	AppendEnumSuffixForSimpleUnionEnums bool
+
```

Comment:
Where are these fields populated? They should be populated here and then handed in to the consumers of the IR, rather than implemented here and populated from downstream consumers.

Per https://github.com/openconfig/ygot/blob/master/docs/code-generation-design.md -- the intention is that the IR provides means to not require any downstream parsing of YANG -- and it should really be considered immutable by downstream consumers.